src/slotscheck/discovery.py

```diff
@@ -299,15 +300,19 @@ def _is_package(p: AbsPath) -> bool:
     return p.is_dir() and (p / _INIT_PY).is_file()
 
 
-def find_modules(p: AbsPath) -> Iterable[ModuleLocated]:
+def find_modules(
+    p: AbsPath, python_path: Optional[FrozenSet[AbsPath]] = None
+) -> Iterable[ModuleLocated]:
     "Recursively find modules at given path. Nonexistent Path is ignored"
+    if python_path is None:
+        python_path = frozenset(map(Path, sys.path))
     if p.name == _INIT_PY:
-        yield from find_modules(p.parent)
+        yield from find_modules(p.parent, python_path)
     elif _is_module(p):
-        parents = [p] + list(takewhile(_is_package, p.parents))
+        parents = [p, *takewhile(lambda p: p not in python_path, p.parents)]
```

Comment:
We should probably explicitly detect and raise on the (unlikely) case is which none of the parents are in `python_path`

Reply:
Done.